### PR TITLE
add OnShipDeathStarted scripting hook

### PR DESCRIPTION
**Pull request by Goober5000**
This occurs at the beginning of the ship death routine, similarly to OnDepartureStarted occurring at the beginning of the ship departure routine.

### CODE REVIEW

**Comment by BMagnu** [comment]
What does this hook gain over the existing OnDeath hook, just by being run a bit earlier in the function?

**Comment by Goober5000** [comment]
It allows just-in-time configuration of certain ship options: persona lament/praise messages, the death message displayed to the player, the vaporize flag, the mission log, etc.  The existing hook at the end of the function happens too late to affect any of that.

**Comment by BMagnu** [comment]
I see, that makes sense.

File: code/mission/missionparse.cpp [hunk 1]
```diff
@@ -7316,10 +7316,10 @@ int mission_do_departure(object *objp, bool goal_is_to_warp)
 
 	mprintf(("Entered mission_do_departure() for %s\n", shipp->ship_name));
 
-	// add scripting hook for 'On Depature Started' --wookieejedi
-	// hook is placed at the begining of this function to allow the scripter to
+	// add scripting hook for 'On Departure Started' --wookieejedi
+	// hook is placed at the beginning of this function to allow the scripter to
 	// actually have access to the ship's departure decisions before they are all executed
-	OnDepartureStartedHook->run(scripting::hook_param_list(scripting::hook_param("Ship", 'o', objp)));
+	OnDepartureStartedHook->run(scripting::hook_param_list(scripting::hook_param("Self", 'o', objp), scripting::hook_param("Ship", 'o', objp)));
```
Comment (BMagnu) [inline code/mission/missionparse.cpp:7322]: If you add a new parameter here (even though this new "Self" parameter seems identical to the ship parameter), it should be documented in the hook itself.

**Review by BMagnu — CHANGES_REQUESTED** [review]

File: code/mission/missionparse.cpp (hunk 1 above)
Comment (Goober5000) [reply code/mission/missionparse.cpp:7322]: Good point

**Review by BMagnu — APPROVED** [review]

**Pull request merged**